### ISSUE TITLE
Doesn't work for private repositories

### ISSUE THREAD

**Issue by manparvesh** [harshjv/github-repo-size#2]
A very helpful extension, @harshjv! But unfortunately it does not seem to work for private repositories. 
It would be great if the size would be visible to the owner.
I think Github authentication would be required in the extension, but it will be worth it.

**Comment by harshjv** [comment]
@manparvesh I am glad that you liked it :)

Ya, currently it won't work with private repositories.

Sure, GitHub auth is required in order to do that. Will work on it next weekend.

Meanwhile, you can always fork it and send a PR for that :)

**Comment by harshjv** [comment]
This looks interesting: https://github.com/michael/github

**Comment by jonathancounihan** [comment]
I have got this work using a authorisation header and a base64 encoded string of the username and a personal access token (https://github.com/settings/tokens)

Any suggestions for the UI to capture this info? Simple popup to capture the username and password/token and then store the base64 string?

I have created a chrome extension before so I am aware of the storage possibilities, but I'm looking for some UI input?
